nomad/structs/structs.go

```diff
@@ -581,6 +581,11 @@ type WriteMeta struct {
 type NodeRegisterRequest struct {
 	Node      *Node
 	NodeEvent *NodeEvent
+
+	// CreateNodePool is used to indicate that the node's node pool should be
+	// create along with the node registration if it doesn't exist.
+	CreateNodePool bool
```

Comment:
I think defaulting to `false` is the safe option here because it will keep the node in the `initializing` status with the node event as a hint to users.

If this were default to `true` (like, `SkipNodePoolCreation`) we would risk placing a node in a node pool that is not replicated to other regions.